q2_demux/tests/test_filter.py

```diff
@@ -0,0 +1,183 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import unittest
+import os
+import pandas as pd
+import q2_demux._filter as filter
+
+from qiime2.plugin.testing import TestPluginBase
+from q2_demux._summarize import _PlotQualView
+from q2_types.per_sample_sequences import (
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt)
+
+from qiime2 import Metadata
+
+
+class FilterSamplesTests(TestPluginBase):
+
+    package = 'q2_demux.tests'
+
+    def setUp(self):
+        super().setUp()
+
+        self.data_single = SingleLanePerSampleSingleEndFastqDirFmt(
+            self.get_data_path('filter_samples_single_end/dir_fmt'), mode='r')
```

Comment:
Since `data_single` doesn't outlive this setup, I would not bother assigning to `self` (it doesn't hurt either).